### PR TITLE
Distributed chat system article #2

### PR DESCRIPTION
**Pull request by melmedany**

### CODE REVIEW

File: data/blog/distributed-chat-system/system-components-communication-and-chat-api.md [hunk 1]
```diff
@@ -0,0 +1,1090 @@
...
+- Ready to use Kafka cluster
+- Web interface to manage Kafka cluster
+
+### Technologies
+
+- Docker
+
+### Implementation
+
+In the project root directory, we will create an empty `docker-compose.yml` file and add the following:
```
Comment (DaveBitter) [inline data/blog/distributed-chat-system/system-components-communication-and-chat-api.md:41]: I feel like the Functional Requirements and Technologies paragraphs could do with a bit of writing to improve storytelling.

For example, something like:

_Let's start with the functional requirements. We need this and this for this and this... For technologies we want to use Docker to.... etc etc_

Comment (DaveBitter) [reply data/blog/distributed-chat-system/system-components-communication-and-chat-api.md:41]: Same goes for the other sections in this article with the same structure :)

File: data/blog/distributed-chat-system/system-components-communication-and-chat-api.md [hunk 2]
```diff
@@ -0,0 +1,1090 @@
...
+    protected void process(NewConnectionDTO newConnectionDTO) {
+        LOGGER.debug("Processing new connection: {}", JsonUtil.toJson(newConnectionDTO));
+        UserProfile user = userProfileService.findById(newConnectionDTO.getRequesterId());
+        UserProfile connection = userProfileService.findById(newConnectionDTO.getConnectionId());
+        conversationService.createConversation(user, connection);
+    }
+}
+```
+
+And `NewConnectionProducer.java`:
```
Comment (DaveBitter) [inline data/blog/distributed-chat-system/system-components-communication-and-chat-api.md]: I feel that all of these code examples are a bit of "code dumping" in the article. Is all the code needed? And if so, can you add what it does and why it works that way?

Comment (DaveBitter) [reply data/blog/distributed-chat-system/system-components-communication-and-chat-api.md]: This goes for all the sections like this :)

Comment (melmedany) [reply data/blog/distributed-chat-system/system-components-communication-and-chat-api.md]: Improved code explanation to not make it feel like 'code dump'. The only thing that is better than that will be going over it line by line 😂

File: data/blog/distributed-chat-system/system-components-communication-and-chat-api.md (hunk 1 above)
Comment (melmedany) [reply data/blog/distributed-chat-system/system-components-communication-and-chat-api.md:41]: I believe that is thoroughly covered in laters sections while exploring each functionality in separate, right?

**Pull request merged**